core/controller/src/main/scala/whisk/core/controller/GatewayRoutesAPI.scala

```diff
@@ -0,0 +1,197 @@
+            val key = Await.result(WhiskAuth.get(authStore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+            val validCredentials = BasicHttpCredentials(key.authkey.uuid.toString, key.authkey.key.toString)
+            val pipeline: HttpRequest => Future[HttpResponse] = (
+                addCredentials(validCredentials)
+                    ~> sendReceive
+            )
+            val p = Promise[String]()
+            val response: Future[HttpResponse] = pipeline(Post(url.withPath(actionPath).toString.concat("?blocking=true"), HttpEntity(ContentTypes.`application/json`, requestBody))  )
+            response.onComplete({
+                case Success(result: HttpResponse) => {
+                    p trySuccess result.entity.asString
+                }
+                case Failure(error) => {
+                    p tryFailure error
+                }
+            })
+            p.future
+        }
+        /**
+        * Handles invoking delete action
+        **
+        * @return Future[String] from action result
+        */
+        def invokeActionDelete(requestBody: String ) : Future[String]  = {
+            val url = Uri(s"http://localhost:${config.servicePort}")
+            val actionPath = Path("/api/v1") / "namespaces" / "whisk.system" / "actions" / "routemgmt" / "deleteApi"
+            val key = Await.result(WhiskAuth.get(authStore, Subject("whisk.system"),false)(TransactionId.unknown), 5.seconds)
+            val validCredentials = BasicHttpCredentials(key.authkey.uuid.toString, key.authkey.key.toString)
+            val pipeline: HttpRequest => Future[HttpResponse] = (
+                    addCredentials(validCredentials)
+                             ~> sendReceive
+            )
+            val p = Promise[String]()
+            val response: Future[HttpResponse] = pipeline(Post(url.withPath(actionPath).toString.concat("?blocking=true"), HttpEntity(ContentTypes.`application/json`, requestBody))  )
+            response.onComplete({
+                case Success(result: HttpResponse) => {
+                   p trySuccess result.entity.asString
+                }
+                case Failure(error) => {
+                   p tryFailure error
+                }
+            })
+            p.future
+      }
+      /**
+      * Handles the controller's api gateway routes
+      */
+      def routes(user: Identity)(implicit transid: TransactionId) = {
+       lazy val collectionPrefix = pathPrefix( (EntityPath.DEFAULT.toString.r | Segment) / "routes")
+       get {
+         collectionPrefix { segment => {
+           entity(as[String]) { pattern =>
+             parameters("basepath", "operation"?"","relpath"?"") { (basepath, operation,relpath) =>
+               var postBody = "{" + "\"namespace\":" +"\"" + segment + "\"," + "\"basepath\":" +"\"" + basepath + "\"," + "\"operation\":" +"\"" + operation + "\"," + "\"relpath\":" +"\"" + relpath + "\"}";
```

Comment:
don't construct the content as a string - this is error prone. You can use a type that you serialize to Json or use a Map that you can serialize to a JsObject. See the spray.json package.